src/validation/DCARegistry.sol

```diff
@@ -0,0 +1,271 @@
+import {ECDSA} from "openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+import {EIP712} from "openzeppelin-contracts/utils/cryptography/EIP712.sol";
+
+/// @notice Registry for tracking and validating DCA order execution with signature verification
+contract DCARegistry is IDCARegistry, IValidationCallback, EIP712 {
+    using ECDSA for bytes32;
+
+    mapping(bytes32 => DCAExecutionState) public executionStates;
+    mapping(address => mapping(uint256 => bool)) public usedNonces;
+    mapping(bytes32 => bool) public usedOrderNonces;
+
+    // Track registered intents to prevent replay attacks
+    mapping(bytes32 => bool) public registeredIntents;
+
+    error InvalidDCAFrequency();
+    error InvalidDCAChunkSize();
+    error DCAFloorPriceNotMet();
+    error InvalidDCAParams();
+    error InvalidSignature();
+    error InvalidCosignature();
+    error IntentExpired();
+    error IntentAlreadyRegistered();
+    error IntentNotRegistered();
+    error NonceAlreadyUsed();
+    error OrderNonceAlreadyUsed();
+    error InvalidTokens();
+    error InvalidCosigner();
+    error InvalidauthorizationTimestamp();
+    error InvalidGasPrice();
+
+    /// @notice EIP-712 type hash for DCA intent
+    bytes32 public constant DCA_INTENT_TYPEHASH = keccak256(
+        "DCAIntent(address inputToken,address outputToken,address cosigner,uint256 minFrequency,uint256 maxFrequency,uint256 minChunkSize,uint256 maxChunkSize,uint256 minOutputAmount,uint256 maxSlippage,uint256 deadline,uint256 nonce,bytes32 privateIntentHash)"
+    );
+
+    constructor() EIP712("DCARegistry", "1") {}
+
+    /// @inheritdoc IValidationCallback
+    function validate(address, ResolvedOrder calldata order) external override {
+        // Decode DCA validation data from additionalValidationData
+        if (order.info.additionalValidationData.length == 0) {
+            revert InvalidDCAParams();
+        }
+
+        bytes memory dcaData = order.info.additionalValidationData;
+
+        // Skip AllowanceTransfer prefix if present
+        if (dcaData.length > 0 && dcaData[0] == 0x01) {
+            // Remove the first byte (AllowanceTransfer flag) and decode the rest
+            bytes memory dcaValidationBytes = new bytes(dcaData.length - 1);
+            for (uint256 i = 1; i < dcaData.length; i++) {
+                dcaValidationBytes[i - 1] = dcaData[i];
+            }
+            dcaData = dcaValidationBytes;
+        }
+
+        DCAValidationData memory validationData = abi.decode(dcaData, (DCAValidationData));
+        DCAIntent memory intent = validationData.intent;
+        DCAOrderCosignerData memory cosignerData = validationData.cosignerData;
+
+        // Calculate intent hash
+        bytes32 intentHash = hashDCAIntent(intent);
+
+        // Verify swapper signature if intent not already registered
+        if (!registeredIntents[intentHash]) {
+            _verifyIntentSignature(intent, validationData.signature, order.info.swapper);
```

Comment:
Is the only benefit of registeredIntents to save us from reverfying the signature?